site/app/models/gradeable/Gradeable.php

```diff
@@ -1541,13 +1541,23 @@ private function checkPath(string $path) {
         while ($checked_paths <= 1000 && count($cur_paths) !== 0) {
             foreach ($cur_paths as $cur_path) {
                 $is_dir = is_dir($cur_path);
-                if (!$this->checkValidPerms($cur_path, $group_map, $user_map, $is_dir)) {
-                    return "Invalid permissions on a file or directory within specified path.";
+
+                $config_upload_dir = $this->core->getConfig()->getCoursePath() . "/config_upload/";
```

Comment:
Recommended to use `FileUtils::joinPaths()`.

```suggestion
                $config_upload_dir = FileUtils::joinPaths($this->core->getConfig()->getCoursePath(), "config_upload");
```